src/tree/VirtualMachineTreeItem.ts

```diff
@@ -45,8 +45,11 @@ export class VirtualMachineTreeItem extends AzureTreeItem {
         return this.virtualMachine;
     }
 
-    public static contextValue: string = 'azVmVirtualMachine';
-    public readonly contextValue: string = VirtualMachineTreeItem.contextValue;
+    public static linuxContextValue: string = 'linuxVirtualMachine';
+    public static windowsContextValue: string = 'windowsVirtualMachine';
+    public static regexpContextValue: RegExp = /VirtualMachine$/;
```

Comment:
A name like `allOsContextValue` may fit better here. But do what sounds good to you I don't feel too strongly about it